src/components/Block/SpaceFormProfile.vue

```diff
@@ -16,67 +16,87 @@ const props = withDefaults(
   (e: 'errors', value: any);
   (e: 'pick', field: any);
+  (e: 'no-network');
 }>();
 
 const availableNetworks = enabledNetworks.map(id => ({
   id,
   name: getNetwork(id).name
 }));
 
-const definition = {
-  type: 'object',
-  title: 'Space',
-  additionalProperties: false,
-  required: ['name', 'walletNetwork', 'walletAddress'],
-  properties: {
-    name: {
-      type: 'string',
-      title: 'Name',
-      minLength: 1,
-      examples: ['Space name']
-    },
-    description: {
-      type: 'string',
-      format: 'long',
-      title: 'About',
-      examples: ['Space description']
-    },
-    externalUrl: {
-      type: 'string',
-      title: 'Website',
-      examples: ['Website URL']
-    },
-    github: {
-      type: 'string',
-      title: 'GitHub',
-      examples: ['GitHub handle']
-    },
-    twitter: {
-      type: 'string',
-      title: 'Twitter',
-      examples: ['Twitter handle']
-    },
-    discord: {
-      type: 'string',
-      title: 'Discord',
-      examples: ['Discord handle']
-    },
-    walletNetwork: {
-      type: 'string',
-      enum: enabledNetworks,
-      options: availableNetworks,
-      title: 'Treasury network'
-    },
-    walletAddress: {
-      type: 'string',
-      format: 'address',
-      title: 'Treasury address',
-      examples: ['0x0000…']
+const definition = computed(() => {
+  return {
+    type: 'object',
+    title: 'Space',
+    additionalProperties: false,
+    required: ['name', 'walletNetwork', 'walletAddress'],
+    properties: {
+      name: {
+        type: 'string',
+        title: 'Name',
+        minLength: 1,
+        examples: ['Space name']
+      },
+      description: {
+        type: 'string',
+        format: 'long',
+        title: 'About',
+        examples: ['Space description']
+      },
+      externalUrl: {
+        type: 'string',
+        title: 'Website',
+        examples: ['Website URL']
+      },
+      github: {
+        type: 'string',
+        title: 'GitHub',
+        examples: ['GitHub handle']
+      },
+      twitter: {
+        type: 'string',
+        title: 'Twitter',
+        examples: ['Twitter handle']
+      },
+      discord: {
+        type: 'string',
+        title: 'Discord',
+        examples: ['Discord handle']
+      },
+      walletNetwork: {
+        type: ['string', 'null'],
+        enum: [null, ...enabledNetworks],
+        options: [{ id: null, name: 'No treasury' }, ...availableNetworks],
+        title: 'Treasury network',
+        nullable: true
+      },
+      walletAddress:
+        props.form.walletNetwork === null
+          ? {
+              type: 'null',
+              title: 'Treasury address',
+              examples: ['0x0000…']
+            }
+          : {
+              type: 'string',
+              title: 'Treasury address',
+              examples: ['0x0000…'],
+              format: 'address',
+              minLength: 1,
+              hidden: props.form.walletNetwork === null
+            }
```

Comment:
At this point I'm thinking that maybe we should just add `walletAddress` conditionally (it either is there, or it's not), and we can just toggle `additionalProperties` to true. With that we wouldn't need `hidden`, `type: 'null'` and all those extra changes - with benefit of being simpler and not adding definitions that basically mean: "nothing".